internal/graphicsdriver/metal/ca/ca_darwin.go

```diff
@@ -49,40 +45,51 @@ type Layer interface {
 //
 // Reference: https://developer.apple.com/documentation/quartzcore/cametallayer.
 type MetalLayer struct {
-	metalLayer unsafe.Pointer
+	metalLayer objc.ID
 }
 
+var (
+	coreGraphics                = purego.Dlopen("/System/Library/Frameworks/CoreGraphics.framework/Versions/Current/CoreGraphics", purego.RTLD_GLOBAL)
+	_CGColorSpaceCreateWithName = purego.Dlsym(coreGraphics, "CGColorSpaceCreateWithName")
+	_CGColorSpaceRelease        = purego.Dlsym(coreGraphics, "CGColorSpaceRelease")
+	kCGColorSpaceDisplayP3      = purego.Dlsym(coreGraphics, "kCGColorSpaceDisplayP3")
+)
+
 // MakeMetalLayer creates a new Core Animation Metal layer.
 //
 // Reference: https://developer.apple.com/documentation/quartzcore/cametallayer.
 func MakeMetalLayer() MetalLayer {
-	return MetalLayer{C.MakeMetalLayer()}
+	layer := objc.ID(objc.GetClass("CAMetalLayer")).Send(objc.RegisterName("new"))
+	if !cocoa.IsIOS {
```

Comment:
I think this constant should be defined in ca as long as this is the only one usage, but what do you think?

Reply:
It's also used in mtl package

Reply:
NVM, I found it https://github.com/hajimehoshi/ebiten/pull/2243/files#diff-f436c80b858b5bc84325c6aa909731c966b62551cefefc37b6f6936ab48dccd5R584